### PR TITLE
Clean up some PROTOTYPE markers in NullableWalker

### PR DESCRIPTION
**Pull request by jcouv**
Here's another batch.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 1]
```diff
@@ -36696,6 +36762,30 @@ static void F(Enumerable e)
...
+        {
+        }
+
+        if (parameter == null) throw null;
+        foreach (T local2 in new[] { parameter })
+        {
+        }
+    }
+}";
+            // https://github.com/dotnet/roslyn/issues/29850 Expecting a W warning for assigning possibly null value to local
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >// https://github.com/dotnet/roslyn/issues/29850 Expecting a W warning for assigning possibly null value to local [](start = 12, length = 113)

This shouldn't be a warning since `local` has the same nullability as `parameter`.

File: src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs [hunk 2]
```diff
@@ -97,7 +96,7 @@ internal sealed class VariableState
         /// </summary>
         private static readonly TypeSymbolWithAnnotations _invalidType = TypeSymbolWithAnnotations.Create(ErrorTypeSymbol.UnknownResultType);
 
-        private TypeSymbolWithAnnotations _resultType; // PROTOTYPE(NullableReferenceTypes): Should be return value from the visitor, not mutable state.
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs]: Are we just not going to address this?

File: src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs [hunk 3]
```diff
@@ -4495,8 +4470,8 @@ internal class LocalState : AbstractLocalState
         internal struct LocalState : AbstractLocalState
 #endif
         {
-            // PROTOTYPE(NullableReferenceTypes): Consider storing nullability rather than non-nullability
-            // or perhaps expose as nullability from `this[int]` even if stored differently.
+            // Consider storing nullability rather than non-nullability
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs]: We should open an issue to track, rather than just removing the word prototype. Otherwise we'll forget this.

Comment (cston) [reply src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs]: Changing the storage does not need to be fixed so I think it's reasonable to use a comment without an associated bug.

---
In reply to: [217797157](https://github.com/dotnet/roslyn/pull/29875#discussion_r217797157) [](ancestors = 217797157)

File: src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs (hunk 2 above)
Comment (jcouv) [reply src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs]: Correct, that's what I suspect at this point.

**Comment by jcouv** [comment]
@cston @dotnet/roslyn-compiler for a second review. Thanks

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for a second review. Thanks

File: src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs [hunk 4]
```diff
@@ -3697,7 +3672,7 @@ private void VisitMemberAccess(BoundExpression receiverOpt, Symbol member, bool
                 {
                     // We are supposed to track information for the node. Use whatever we managed to
                     // accumulate so far.
-                    if (resultType.IsReferenceType)
+                    if (!resultType.IsValueType)
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs]: >if (!resultType.IsValueType) [](start = 20, length = 28)

It is not obvious why this change is necessary.

Comment (jcouv) [reply src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs]: Here's the [corresponding test](https://github.com/dotnet/roslyn/pull/29875/commits/a26c373c8d63af4afc00ed1804ef3d7c8dddab11#diff-4ef499b331bacc63b10e31e95a00db61): `Members_Fields_UnconstrainedType`
